src/TypeSpec.Extension/Emitter.Csharp/src/lib/utils.ts

```diff
@@ -46,44 +40,21 @@ export function getNameForTemplate(model: Model): string {
     return model.name;
 }
 
-export function getProjectedNameForCsharp(
-    context: SdkContext,
-    type: Type
-): string | undefined {
-    const projectedNamesMap = getProjectedNames(context.program, type);
-    return (
-        projectedNamesMap?.get(projectedNameCSharpKey) ??
-        projectedNamesMap?.get(projectedNameClientKey)
-    );
-}
-
 export function getTypeName(
     context: SdkContext,
     type: Model | Enum | EnumMember | ModelProperty | Scalar | Operation
 ): string {
-    var name =
-        getProjectedNameForCsharp(context, type) ??
-        getFriendlyName(context.program, type);
-    if (name) return name;
-    if (type.kind === "Model") {
-        name = getNameForTemplate(type);
-        if (name === "") {
+    var name = getLibraryName(context, type);
+    if (type.kind !== "Model") return name;
+    if (type.name === name) {
+        var templateName = getNameForTemplate(type);
+        if (templateName === "") {
             const sdkModel = getSdkModel(context, type as Model);
-            return sdkModel.generatedName || sdkModel.name;
+            return sdkModel[0].generatedName || sdkModel[0].name;
```

Comment:
Can we change to:

const [sdkModel] = getSdkModel(context, type as Model);
return sdkModel.generatedName || sdkModel.name;